migrations/2_deploy_contracts.js

```diff
@@ -1,55 +1,63 @@
-  			
-  			const startTime = web3.eth.getBlock('latest').timestamp  + 60;
-  			//10 minutes later
-		  	const endTime = startTime + (60 * 10);
-
-		  	const rate = new web3.BigNumber(1000);
-		  	const goal = new web3.BigNumber(web3.toWei(300, 'ether'));
-		  	const cap = new web3.BigNumber(web3.toWei(300, 'ether'));
-		  	const wallet = "0x627306090abab3a6e1400e9345bc60c78a8bef57";
-		  	
-
-  			return deployer.deploy(RSTCrowdsale, startTime, endTime, rate, goal, cap, wallet, ResearchSpecificToken.address).then(_ =>{
-  				const token = ResearchSpecificToken.at(ResearchSpecificToken.address);
-  				token.transferOwnership(RSTCrowdsale.address).then( __ =>{
-  					console.log("Deployment complete!");
-  				}).catch( e =>{
-  					console.log("Cant transfer token ownership", e);
-  				});
-
-  				
-
-  			 }).catch( e =>{
-  				console.log("RSTCrowdsale Deployment failed", e);
-  			});
-  	     
-  		 }).catch( e =>{
-  			console.log("ResearchSpecificToken Deployment failed", e);
-  		});
-  		*/  		
-  }).catch( e =>{
-  		console.log("IkuToken Deployment failed", e);
-  });
-  
-}
+	var registry;
+	var ikuToken, researchSpecificToken;
+	var ikuTokenProxy, researchSpecificTokenProxy;
+
+	// only deploy registry contracts once
+	deployer.deploy(Registry);
+
+	//deploy all version-1 contracts
+	deployer.deploy(IkuToken);
+	deployer.deploy(
+		ResearchSpecificToken,
+		18,
+		'ResearchSpecificToken',
+		'RST1'
+	);
+
+	// step 1 : register deployed version-1 contracts with the registry contract
+	// step 2 : creating proxies
+	// step 3 : fetch proxy addresses 
+	deployer.then(() => {
+		return Registry.deployed();
+	}).then(instance => {
+		registry = instance;
+		return IkuToken.deployed();
+	}).then(instance => {
+		ikuToken = instance;
+		return ResearchSpecificToken.deployed();
+	}).then(instance => {
+		researchSpecificToken = instance;
+		return registry.addVersion('IkuToken', '1.0', ikuToken.address);
+	}).then(tx => {
+		return registry.createProxy('IkuToken', '1.0');
+	}).then(tx => {
+		ikuTokenProxy = tx.logs[0].args.proxy;
+		console.log('IkuToken proxy: ', ikuTokenProxy);
+		return registry.addVersion(
+	      'ResearchSpecificToken',
+	      '1.0',
+	      researchSpecificToken.address
+	    );
+	}).then(tx => {
+		return registry.createProxy('ResearchSpecificToken', '1.0');
+	}).then(tx => {
+		researchSpecificTokenProxy = tx.logs[0].args.proxy;
+		console.log('ResearchSpecificToken proxy: ', researchSpecificTokenProxy);
+	});
+
+	// step 4 : call any uncalled initializers
+	// Note : there's no need to call initialize on Ikutoken because it's 
+	//        automatically called by registry on proxy creation.
+	deployer.then(()=>{
+		return ResearchSpecificToken.at(researchSpecificTokenProxy);
+	}).then((instance)=>{
+		researchSpecificToken = instance;
+		console.log('initializing ResearchSpecificToken ...');
+		return researchSpecificToken.initializeRSToken(18, 'ResearchSpecificToken', 'RST')
```

Comment:
The ResearcSpecificToken should be deployed only via jobs. So this shouldn't be here.